src/Surfnet/StepupRa/RaBundle/Security/Authentication/Provider/SamlProvider.php

```diff
@@ -69,6 +80,14 @@ public function __construct(
      */
     public function authenticate(TokenInterface $token)
     {
+        if ($token->assertion->getIssuer() !== $this->remoteIdp) {
```

Comment:
The authentication provider is the place where a token is authenticated; we check whether we have a identity that corresponds to the information in the assertion. It looks like the `ProcessSamlAuthenticationHandler` is a more logical place; we process all kinds of preconditions here (like whether the request ID matches). (It's getting a bit out of hand there, but that's another subject.) Even though we only have one remote IdP, we could store the expected IdP in the `SamlAuthenticationStateHandler` when we initiate the SAML interaction in `SamlInteractionProvider`. This value can then be compared to the Issuer. This way we don't need an extra configuration value whose purpose may need explanation/documentation, and it Just Works™.